[Docs] Tidy Vizro-ai docs (#466)
mckinsey/vizro: vizro-ai/docs/pages/user-guides/add-generated-chart-usecase.md

# How to add Vizro-AI created charts into a Vizro dashboard

> This guide explains different ways in which you can add a chart generated by Vizro-AI to your [Vizro](https://github.com/mckinsey/vizro/tree/main/vizro-core) dashboard.

Reviewer: ```suggestion
This guide explains the different ways in which you can add a chart generated by Vizro-AI to your [Vizro](https://github.com/mckinsey/vizro/tree/main/vizro-core) dashboard.
```

> ###  Access chart-generating code

Reviewer: ```suggestion
##  Access chart-generating code
```

Reviewer: I went back and forth with the wording here but I think the elements are that you are using the code that Vizro-AI generates, so it's best to explain that simply. It means the word "Chart" is omitted. An alternative would be "Use generated code to create the chart" or "Use Vizro-AI generated code to create the chart" but that gets quite wordy. Which do you prefer?

Author: I actually prefer the committed version "Use Vizro-AI's generated code". Its simple and it is obvious from the introduction above that we are trying to generate a chart using VizroAI. Or do you prefer the examples with the word chart in?

Reviewer: I don't think we need the word chart here, let's go with your preference.

    [VizroDashboard]: ../../assets/user_guides/chart_into_dashboard_large.gif

> ### Direct use of `fig` object

Reviewer: ```suggestion
### Use Vizro-AI dynamically to return a `fig` object
```

> Additionally, we can assign the output of Vizro-AI `plot()` method directly to the fig variable, enabling its reuse in the `vm.Graph.figure` argument.

Reviewer: ```suggestion
Alternatively, we can use Vizro-AI dynamically and assign the output of `plot()` directly to the fig variable, enabling its reuse in the `vm.Graph.figure` argument.
```

This method offers streamlined efficiency, eliminating the need for code copying.

> However, it is important to note that each dashboard run triggers an API call to OpenAI, potentially escalating costs, but this only occurs if the fig object is not stored and is re-run by a Vizro-AI `plot()` call.

Reviewer: ```suggestion
However, it is important to note that each dashboard run triggers an API call to OpenAI, potentially escalating costs, although this can be avoided if the `fig` object is stored and retrieved as needed, instead of making repeated calls to `plot()`.
```

> Executing the code below will yield the identical dashboard as the initial example.

Reviewer: ```suggestion
Executing the code below yields the identical dashboard as the example above.
```

    [VizroDashboard]: ../../assets/user_guides/chart_into_dashboard_large.gif

### Direct use of `fig` object

Additionally, we can assign the output of Vizro-AI `plot()` method directly to the fig variable, enabling its reuse in the `vm.Graph.figure` argument.

> This method offers streamlined efficiency, eliminating the need for code copying. However, it is important to note that each dashboard run triggers an API call to the OpenAI, potentially escalating costs.

Reviewer 2: it might be a bit misleading here if we say each dashboard run triggers an API call to openai. In theory if fig is returned with `fig = vizro_ai.plot(df, xx)`, fig is returned as fig object, and only restarting dashboard won't trigger cost.

added suggestion as below

> This method offers streamlined efficiency, eliminating the need for code copying. However, it is important to note that each dashboard run triggers an API call to the OpenAI, potentially escalating costs.

Reviewer 2: ```suggestion
This method offers streamlined efficiency, eliminating the need for code copying. However, it is important to note that each dashboard run triggers an API call to OpenAI, potentially escalating costs, but this only occurs if the fig object is not stored and is re-run by a Vizro-AI `plot()` call.
```
